library/include/DataFuncs.h

```diff
@@ -34,229 +34,106 @@ distribution.
 
 namespace DFHack {
     class color_ostream;
+
+    namespace Lua {
+        template<typename T>
+        T Get(lua_State *L, int idx) {
+            T val;
+            df::identity_traits<T>::get()->lua_write(L, UPVAL_METHOD_NAME, &val, idx);
+            return val;
+        }
```

Comment:
i concur, there doesn't seem to be a reason why this needs to be in the `DFHack::Lua` namespace

Reply:
I agree that it would be nice to have this in `LuaTools.h`. This is the reason why I put it in the namespace where it currently is. The reason I didn't put it to the right header immediately is that this would require `DataFuncs` to import `LuaTools.h` and IIRC `LuaTools.h` to include `DataIdentity`.h , and I wasn't sure about the relationship between the various Lua headers.